app/core/config.py

```
@@ -19,6 +22,144 @@ class LogLevel(str, Enum):
+        return v
+    
+    @field_validator("MAIL_TEMPLATES_PARENT_DIR", mode="after")
+    @classmethod
+    def _normalize_parent(cls, v: Path | None) -> Path | None:
+        return v.expanduser().resolve() if isinstance(v, Path) else v
+    
+    @model_validator(mode="after")
+    def _validate(self) -> "MailSettings":
+        # TLS mode sanity
+        if self.MAIL_STARTTLS and self.MAIL_SSL_TLS:
+            raise ValueError("Set only one of MAIL_STARTTLS or MAIL_SSL_TLS, not both.")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)